lib/tad/tad_pkt.c

```diff
@@ -609,7 +610,7 @@ tad_pkt_flatten_copy(tad_pkt *pkt, uint8_t **data, size_t *len)
 
     total_len = (len == NULL || *len == 0) ? tad_pkt_len(pkt) : *len;
     if (*data == NULL)
-        *data = TE_ALLOC(total_len);
+        *data = TE_ALLOC((total_len == 0 && !warn_empty_pkt) ? 1 :total_len);
```

Comment:
IMHO solution is really artificial. We should not allocate memory if requested length is 0. That's it. And there is no need for extra variable to control warning.